docs/decisions/0001-meilisearch.rst

```diff
@@ -0,0 +1,146 @@
+Meilisearch, rather than routing them through Django. Initial testing has shown
+it to be much more developer friendly than Elasticsearch/OpenSearch.
+
+At the time of writing, there are only two known concerns with Meilisearch:
+
+1. It doesn't (yet) support High Availability via replication, although this is
+   planned and under development. It does have other features to support high
+   availability, such as very low restart time (in ms).
+2. It doesn't support boolean operators in keyword search ("red AND panda"),
+   though it does of course support boolean operators in filters. This is a
+   product decision aimed at keeping the user experience simple, and is unlikely
+   to change.
+
+Problem with edx-search
+=======================
+
+The edx-search feature is currently utilized only within the edx-platform, and many of its implementations are now deprecated in the new microfrontend environments. Additionally, it lacks the personalized token functionality, making it incompatible with advanced search engines such as Meilisearch. Even if we attempt to extend edx-search to support Meilisearch, it would still follow the same proxy mechanism and would not be able to utilize personalized tokens effectively. To find a detailed list of architectural issues here.
+
+Open edX Search API
+===================
+
+We are proposing a new abstraction package with more generalised architecture inorder to support all available search engines including elasticsearch, meilisearch and algolia.
+The Open edX Search API is designed with a forward-thinking approach, ensuring seamless integration with a wide range of future search engines.
+This package aims to provide flexibility, scalability, and ease of use for developers looking to enhance their search capabilities within the Open edX platform.
+
+
+Architecture
+============
+
+The Open edX Search API comprises four main components: the Driver, Indexer, Load Indexes Command, and a JavaScript library. Each component is described in detail below.
+
+Driver Class
+============
+
+The base driver class contains the following functions: `get_user_token()`, `check_connection()`, `indexes()`, `index()`, and `get_search_rules()`. To integrate a new search engine or driver, the child class must implement all these functions.
+This class is responsible for initializing the communication protocol between client and the search service. To change the driver class by updating the value of `SEARCH_ENGINE` in django settings.
+the default value is `SEARCH_ENGINE = "openedx_search_api.drivers.meilisearch.MeiliSearchEngine"`
+
+Indexer Class
+=============
+
+The base indexer class consists of the `index()` and `index_documents()` functions. To override the default functionality of the base class, the child indexer class should implement these functions.
+Inorder to update the indexer class set `INDEXER_CLASS` attribute in django settings default value is `INDEXER_CLASS = "openedx_search_api.indexers.base.BaseIndexer"`
```

Comment:
Why do we need an indexer separate from the driver? Your driver code already has a

```
def index(self, index_name, payload):
        """
        Index the provided payload in the specified index.
```
which seems to do the same thing as the indexer. Could the `index_document` and `index_documents` functions just be part of the driver? Also, how do we remove a document from the index?

Reply:
Index return object of index and index_documents actually push object to search engine.